ui/src/app/components/webcomponents/action-step-form.component.ts

```diff
@@ -1921,7 +1921,8 @@ export class ActionStepFormComponent extends BaseComponent implements OnInit {
         this.assignDataValue(this.getDataTypeString(TestDataType.parameter, data));
       }
       else if( typeof(data)==="object" ){
-        this.testStep.testDataProfileStepId = data?.testDataProfileStepId;
+        if(this.testStep.type===TestStepType.FOR_LOOP)
+          this.testStep.testDataProfileStepId = data?.testDataProfileStepId;
```

Comment:
@nixonakash01 Ragul has some changes related to FOR LOOP and I would suggest you to re verify this after his PR is merged

Reply:
okay bro